tests/pebble.rs

```diff
@@ -334,13 +265,169 @@ impl PebbleEnvironment {
         Ok(roots)
     }
 
-    fn challenge_management_url(&self) -> &str {
-        "http://127.0.0.1:8055" // Default.
+    fn challenge_management_url(&self) -> String {
+        format!("http://[::1]:{}", self.config.challtestsrv_port)
     }
 
     fn directory_url(&self) -> String {
-        format!("https://{}/dir", &self.config.listen_address)
+        format!("https://{}/dir", &self.config.pebble_config.listen_address)
+    }
+}
+
+/// Configuration for running a `PebbleEnvironment` with unique per-environment ports.
```

Comment:
This commit has a lot going on, do you think it could usefully be split into smaller parts?